### PR TITLE
Block Bindings: Remove Block Bindings icon from List View, fixes and CSS updates.

### PR DESCRIPTION
**Pull request by artemiomorales**
## What?
This is a quick follow up to https://github.com/WordPress/gutenberg/pull/59185

- **Removes** the icon from the List View as decided in https://github.com/WordPress/gutenberg/pull/59477#issuecomment-1980371587
- Removes unneeded CSS of the Block Bindings Toolbar indicator.
- Simplifies the SVG of the icon - it uses a single `<path>` now instead of 6 of them.
- [Addresses a console error](https://github.com/WordPress/gutenberg/pull/59185#issuecomment-1973117561). In React, we need to use `fillRule` instead of `fill-rule`.

## Testing Instructions

Please follow the instructions in https://github.com/WordPress/gutenberg/pull/59185

### CODE REVIEW

**Comment by SantosGuillamot** [comment]
I'm not sure what is the desired UI here. I can see the paragraph icon, for example, just turns from black to white. Could something like this make sense when it is selected?

Screenshot 2024-03-01 at 10 04 51

@SaxonF @jasmussen Any thoughts what should be done here?

**Comment by artemiomorales** [comment]
> I'm not sure what is the desired UI here. I can see the paragraph icon, for example, just turns from black to white. Could something like this make sense when it is selected?

@SantosGuillamot Hmm that's not right. It should be showing with the purple outline and a white fill:

Screenshot 2024-02-29 at 2 54 02 PM

I've tested in a different environment and it's appearing as expected. Are you seeing this in the post editor, site editor, or both? Can you clear your cache and restart your environment to see if the issue persists?

**Comment by SantosGuillamot** [comment]
Oh, sorry. I was just suggesting another approach that came to my mind based on the current behavior of other icons. My point was that I'd like to get some feedback to ensure the UI we provide is the desired one.

**Comment by artemiomorales** [comment]
> Oh, sorry. I was just suggesting another approach that came to my mind based on the current behavior of other icons. My point was that I'd like to get some feedback to ensure the UI we provide is the desired one.

Ah ok! In that case I'd say that leaving the icon with purple outline and white fill is best, in my opinion.

The blue highlight of the selected block already signals a change, and if we also modify the icon, I believe it introduces cognitive dissonance with too many things changing at once and could look like a mistake with inconsistency of the icon across the UI, though I'd like to hear what design thinks 😄

**Comment by jasmussen** [comment]
Thanks for following up on this. I'd personally love to find some time to explore a couple of more icon variations — the current one is pretty good, but I think we can be even better.

That said, responding to [this comment](https://github.com/WordPress/gutenberg/pull/59477#issuecomment-1972794855) — this is actually what I'd expect: the icon not having color there. So the screenshot there is actually what I'd expect.

**Comment by michalczaplinski** [comment]
> That said, responding to https://github.com/WordPress/gutenberg/pull/59477#issuecomment-1972794855 — this is actually what I'd expect: the icon not having color there. So the screenshot there is actually what I'd expect.

This makes sense. So, the final UI should look like: 

connections

-----

Additionally, I believe the SVG paths need to be adjusted. Currently the storybook for the connection icon looks like this:

Screenshot 2024-03-04 at 13 04 16

**Comment by michalczaplinski** [comment]
@jasmussen Could you provide us with an SVG file that meets those requirements and renders correctly in the storybook? The one in the [Figma file](https://www.figma.com/file/W2SO7yJG09O65q7JjfXZ4m/Block-connections?type=design&node-id=385-5610&mode=design&t=eogT3VAAmA76YKM6-0) is not quite correct.

I tried converting it myself and ask gpt for help but that didn't result in a working code, unfortunately 😢

**Comment by jasmussen** [comment]
Try this one:

```
<svg width="24" height="24" fill="none" xmlns="http://www.w3.org/2000/svg"><path fill-rule="evenodd" clip-rule="evenodd" d="M19.53 4.47a.75.75 0 0 1 0 1.06L17.06 8l.77.769a3.155 3.155 0 0 1 .685 3.439 3.15 3.15 0 0 1-.685 1.022v.001L13.23 17.83v.001a3.15 3.15 0 0 1-4.462 0L8 17.06l-2.47 2.47a.75.75 0 0 1-1.06-1.06L6.94 16l-.77-.769a3.154 3.154 0 0 1-.685-3.439 3.15 3.15 0 0 1 .685-1.023l4.599-4.598a3.152 3.152 0 0 1 4.462 0l.769.768 2.47-2.47a.75.75 0 0 1 1.06 0Zm-2.76 7.7L15 13.94 10.06 9l1.771-1.77a1.65 1.65 0 0 1 2.338 0L16.77 9.83a1.649 1.649 0 0 1 0 2.338h-.001ZM13.94 15 9 10.06l-1.77 1.771a1.65 1.65 0 0 0 0 2.338l2.601 2.602a1.649 1.649 0 0 0 2.338 0v-.001L13.94 15Z" /></svg>
```

I think there are further options to improve it, but we can look at that separately.

**Comment by SantosGuillamot** [comment]
@jasmussen I have just tested the SVG you shared but it seems it is completely white.

If I remove the fill="none" property it looks like this:

Screenshot 2024-03-05 at 09 39 08

It might need to fill the whole SVG, right?

I think it can be fixed removing the `fill:"none", and using `viewbox`. Pushing a commit right now with the code that seems to work for me. Thanks a lot for your help!

**Comment by SantosGuillamot** [comment]
I've updated the SVG in [this commit](https://github.com/WordPress/gutenberg/pull/59477/commits/88850c3c504327073dedc84de6beea0471a2e350). This is how it looks in the storybook right now:

Screenshot 2024-03-05 at 09 44 25

We need to update the CSS logic to match it. Working on it now.

**Comment by jasmussen** [comment]
Try this:

```
/**
 * WordPress dependencies
 */
import { SVG, Path } from '@wordpress/primitives';

const connection = (
	<SVG
		viewBox="0 0 24 24"
		xmlns="http://www.w3.org/2000/svg"
		fill-rule="evenodd"
	>
		<Path
			fillRule="evenodd"
			clipRule="evenodd"
			d="M19.53 4.47a.75.75 0 0 1 0 1.06L17.06 8l.77.769a3.155 3.155 0 0 1 .685 3.439 3.15 3.15 0 0 1-.685 1.022v.001L13.23 17.83v.001a3.15 3.15 0 0 1-4.462 0L8 17.06l-2.47 2.47a.75.75 0 0 1-1.06-1.06L6.94 16l-.77-.769a3.154 3.154 0 0 1-.685-3.439 3.15 3.15 0 0 1 .685-1.023l4.599-4.598a3.152 3.152 0 0 1 4.462 0l.769.768 2.47-2.47a.75.75 0 0 1 1.06 0Zm-2.76 7.7L15 13.94 10.06 9l1.771-1.77a1.65 1.65 0 0 1 2.338 0L16.77 9.83a1.649 1.649 0 0 1 0 2.338h-.001ZM13.94 15 9 10.06l-1.77 1.771a1.65 1.65 0 0 0 0 2.338l2.601 2.602a1.649 1.649 0 0 0 2.338 0v-.001L13.94 15Z"
		/>
	</SVG>
);

export default connection;
```

**Comment by SantosGuillamot** [comment]
> Try this

I tried something similar [here](https://github.com/WordPress/gutenberg/pull/59477#issuecomment-1978234115) and seems to work. Thanks for your help @jasmussen ☺️

**Comment by SantosGuillamot** [comment]
Okay, I've adapted the styles and make some changes based on feedback:

### Before SVG and styles changes

* In block toolbar
Screenshot 2024-03-05 at 10 02 16

* In block list when is selected and not selected
Screenshot 2024-03-05 at 10 02 31

* In storybook
Screenshot 2024-03-05 at 10 07 15

### After SVG and styles changes

* In block toolbar
Screenshot 2024-03-05 at 09 57 52


* In block list when is selected and not selected
Screenshot 2024-03-05 at 09 58 11


* In storybook

Screenshot 2024-03-05 at 10 08 14

---

We can see that, with the new changes, the SVG in the block toolbar looks wider. I know it can be solved with CSS and stroke-width, but I'm not sure if that's the right approach. I'll take a look if it can be solved other way.

Apart from that, I can see that if I enabled the distraction free mode, the icon is not aligned. We must take a look at that as well:

Screenshot 2024-03-05 at 10 10 18

**Comment by jasmussen** [comment]
The icon looks misaligned even in the list view.

I wonder, should we just remove it from the list view for now? There's enough feedback that it isn't clear that's super useful, perhaps this isn't the right approach?

**Comment by SantosGuillamot** [comment]
I added this workaround for the distraction free issue: [link](https://github.com/WordPress/gutenberg/pull/59477/commits/0a157e5972f3fa73bd561711f9f62c9d738a6b7b).

However, I believe we have to explore deeper if we can use core components that already have those styles. For me, ideally, the bindings indicator CSS should be only:

```css
.block-editor-block-bindings-toolbar-indicator svg{
  stroke: var(--wp-bound-block-color);
}
```

And the rest should be managed by core components because it is shared CSS.

For example, if I use `ToolbarButton` instead of `ToolbarItem` at [this point](https://github.com/WordPress/gutenberg/blob/0a157e5972f3fa73bd561711f9f62c9d738a6b7b/packages/block-editor/src/components/block-bindings-toolbar-indicator/index.js#L11), everything works out of the box without extra CSS. I believe we decided not to use `ToolbarButton` because it isn't exactly a button, but we could have something similar I guess. cc: @michalczaplinski

**Comment by SantosGuillamot** [comment]
> should we just remove it from the list view for now? There's enough feedback that it isn't clear that's super useful, perhaps this isn't the right approach?

I'm fine removing it if the general feedback is that it isn't that useful. We can always add it back later.

**Comment by michalczaplinski** [comment]
> For example, if I use ToolbarButton instead of ToolbarItem at [this point](https://github.com/WordPress/gutenberg/blob/0a157e5972f3fa73bd561711f9f62c9d738a6b7b/packages/block-editor/src/components/block-bindings-toolbar-indicator/index.js#L11), everything works out of the box without extra CSS. I believe we decided not to use ToolbarButton because it isn't exactly a button, but we could have something similar I guess. cc: @michalczaplinski

Yes, exactly; we got feedback (https://github.com/WordPress/gutenberg/pull/59185#issuecomment-1959454476) that having a button that does not do anything is not good for a11y. 

I also notice that the icon has a thin black fill:

Screenshot 2024-03-05 at 10 50 42

**Comment by jasmussen** [comment]
Tangential question, was the "select parent" button removed?

**Comment by michalczaplinski** [comment]
> Tangential question, was the "select parent" button removed?

It wasn't. It will appear before the bound block indicator if the bound block has a parent:

Screenshot 2024-03-05 at 11 58 54

**Comment by jasmussen** [comment]
> It wasn't. It will appear before the bound block indicator if the bound block has a parent:

Thank you! CC: @annezazu as I think she encountered some glitches.

**Comment by michalczaplinski** [comment]
I've just pushed a change that I hope fixes the existing issues:

- Fixing the thin black line inside of the icon ( Screenshot 2024-03-05 at 12 29 35 )
- Removing the unneded CSS
- Add the `display: flex` in https://github.com/WordPress/gutenberg/blob/1778ccd0624017f3f2abeb3f9725566bbe04bf19/packages/block-editor/src/components/list-view/style.scss#L579-L581 which should fix the issue in the List View.

Would be great if you could review @SantosGuillamot or @artemiomorales 🙏

**Comment by youknowriad** [comment]
Given that RC1 is today, I'm removing this from 6.5 as non essential.

**Comment by SantosGuillamot** [comment]
> Would be great if you could review @SantosGuillamot or @artemiomorales 🙏

I've tested it and it seems to work as expected. However, I'll wait to approve it until we decide if we should remove the icon from the block list or not.

> Given that RC1 is today, I'm removing this from 6.5 as non essential.

Regarding this, right now there is a console error when you select a block that is connected introduced [here](https://github.com/WordPress/gutenberg/pull/59185#issuecomment-1973117561). Should/could it be fixed in a separate PR for 6.5 @youknowriad? Working on it just in case it is possible.

**Comment by youknowriad** [comment]
> Regarding this, right now there is a console error when you select a block that is connected introduced https://github.com/WordPress/gutenberg/pull/59185#issuecomment-1973117561. Should/could it be fixed in a separate PR for 6.5 @youknowriad? Working on it just in case it is possible.

Yes, I think this one can be sanctioned, can we have an isolated PR/issue for it to be added to the sanctioned items on the board.

**Comment by SantosGuillamot** [comment]
>can we have an isolated PR/issue for it to be added to the sanctioned items on the board.

I have just created a separate PR for that: [link](https://github.com/WordPress/gutenberg/pull/59598). Let me know if that's enough or I should do something else.

**Comment by youknowriad** [comment]
That should be fine. Thank you.

**Comment by michalczaplinski** [comment]
@youknowriad Is there any chance we could still get this in? The majority of the changes are CSS whitespace 🙂 

The icon in the List view currently looks broken:

Screenshot 2024-02-29 at 19 30 01

We can also remove the icon in the ListView altogether.

**Comment by youknowriad** [comment]
Unfortunately no, I have punted several similar small changes as well. It's just that for RC1, it's only critical things that are allowed.

**Comment by michalczaplinski** [comment]
> Unfortunately no, I have punted several similar small changes as well. It's just that for RC1, it's only critical things that are allowed.

I understand, thanks for clarifying! 🙂

**Comment by annezazu** [comment]
On the parent block front, here's what I saw when testing yesterday:

https://github.com/WordPress/gutenberg/assets/26996883/a93e9b87-d5c6-4fe1-8740-980126e344e9

Note: I'm using WP Playground and have run into tons of issues there so perhaps it's a limitation of playground. Tested again using Playground for this PR and WP nightly:

https://github.com/WordPress/gutenberg/assets/26996883/d84c3005-e3f1-431b-960b-72a9f15fbeab

Another item of feedback shared in the video above-- I'd want the icons in this order: image, block binding, lock. Right now, having the image preview shoved between the two in list view is a bit clunky:

Screenshot 2024-03-05 at 9 42 58 AM

**Comment by artemiomorales** [comment]
> On the parent block front, here's what I saw when testing yesterday:

@annezazu This issue with the parent selector not being visible appears to be a pattern overrides issue and I think is best handled separately — pull request created here: https://github.com/WordPress/gutenberg/pull/59621

> Another item of feedback shared in the video above-- I'd want the icons in this order: image, block binding, lock. Right now, having the image preview shoved between the two in list view is a bit clunky

I've swapped the positions of the icons, but I'm not sure the result is any better. Now the connections icon and lock icon appear very far apart due to all of the icons' different sizes and treatment of white space.

Given what @jasmussen mentioned above, perhaps of least resistance is to just remove it for now. Otherwise, Joen do you or perhaps @SaxonF have ideas on how to handle?

Screenshot 2024-03-05 at 9 09 55 PM

**Comment by annezazu** [comment]
We also can keep it as is if design thinks it's better. I didn't realize how odd the spacing would look in general. Something to follow up on outside of this release and refine.

**Comment by SantosGuillamot** [comment]
> Given what @jasmussen mentioned above, perhaps of least resistance is to just remove it for now. Otherwise, Joen do you or perhaps @SaxonF have ideas on how to handle?

In my opinion, we could just remove it until we figure out what is the best strategy for the UX in this regard. It seems it isn't clear if we should have that icon at all.

**Comment by SaxonF** [comment]
That sounds fine. We can revisit after 6.5

File: packages/block-editor/src/components/block-bindings-toolbar-indicator/style.scss
Comment (michalczaplinski) [inline packages/block-editor/src/components/block-bindings-toolbar-indicator/style.scss:6]: We can now remove those properties because of the updates to the <path> of the SVG.

**Comment by michalczaplinski** [comment]
I've removed the Icon from the List View and updated the PR description to reflect that 👍

**Comment by artemiomorales** [comment]
Most of these are formatting changes. Tested the latest changes and it's looking good to me — the list icon is removed, and the icon still appears in the block toolbar as expected.

**Review by michalczaplinski — APPROVED** [review]
LGTM 👍

**Pull request merged**
